[Feature] Enable sorting by tags

SebieF opened the pull request:
Once applied, this PR will make it possible to sort tasks by given tags. 

Tags are sorted in the following order:
```
(no tags)
(no tags)
A B
A C
B
B D
```

I used `nvm install 20` node version for building and testing. I added a unit test and ran it using `npx vitest run`. For translations, I added a german translation. If it is necessary to add all translations before, please add a comment and I will add them soon. I wanted to create the PR first such that I can get feedback if my implemented feature matches the expectations for the project :) 

Closes #1466.

raimund-schluessler commented:
Thanks for the contribution, I will have a look.

Could you please already remove the added translation files? Translation is handled via Transifex, what is included manually will be overwritten at the next sync.

SebieF commented:
You are welcome and thanks for taking a look! I removed the translation files and fixed the linter errors, let me know if any additional changes are necessary :)

raimund-schluessler reviewed (approved):
Works.
